lib/gpu/lib/src/buffer.dart

```diff
@@ -23,6 +23,74 @@ class BufferView {
       {required this.offsetInBytes, required this.lengthInBytes});
 }
 
+/// A buffer that can be referenced by commands on the GPU.
+mixin Buffer {}
```

Comment:
Its difficult for me to provide an alternative solution here, since I don't know what we'll use buffer view for yet - but an empty interface is always a code smell in Dart.

Reply:
> I don't know what we'll use buffer view for yet

`BufferView` is just for binding a section of a buffer to a command, like how Impeller works today. Vertex buffers, index buffers, uniform buffers, SSBOs.

We definitely don't need `HostBuffer` for a minimal solution so I'm considering just removing it for now. Would prefer to do so in a later patch though.

Reply:
This comment is about the `mixin Buffer {}`.

Reply:
Well yes -- the `Buffer` mixin is solely so we can place both kinds of buffers into a `BufferView`, and you said you're not sure how the "buffer view" is going to be used. And we wouldn't need this mixin if we didn't have a `HostBuffer`.